python/setup.py

```diff
@@ -251,8 +251,9 @@ def get_packages(self):
     if RAY_EXTRA_CPP:
         setup_spec.extras["cpp"] = ["ray-cpp==" + setup_spec.version]
 
-    if sys.version_info >= (3, 7, 0):
+    if sys.version_info >= (3, 7):
         setup_spec.extras["k8s"].append("kopf")
+        setup_spec.extras["data"].append("polars >= 0.14.0")
```

Comment:
do we actually want to depend on this here? should this be optional?

Reply:
I made this a `ray[data]` dependency under the assumption that, by the time this is merged, Polars will be the default backend for Arrow-based sorting, groupbys, and aggregations, but I can make this softer to start out.